src/finfield.c

```diff
@@ -529,7 +529,8 @@ FF              FiniteField (
 
     /* search through the finite field table                               */
     l = 1; n = NUM_SHORT_FINITE_FIELDS;
-    while (l <= n) {
+    ff = 0;
+    while (l <= n && SizeFF[l] <= q && q <= SizeFF[n]) {
       /* interpolation search */
```

Comment:
I don't see a need for brackets here, and would think they'd decrease readability here.

I noticed `SIZE_FF` -- not sure why it exists, perhaps for historic reasons? In any case, I followed the classic approach of trying to "blend in" with the immediately surrounding code, which uses `SizeFF`, a lot. I didn't want to start wondering about larger issues when trying to fix a simple localized bug. I don't think it should be done within this PR.